librad/src/identities/git.rs

```diff
@@ -174,6 +182,25 @@ impl<'a, T: 'a> Identities<'a, T> {
         root.verify(progeny)
     }
 
+    pub fn is_fork_generic(&self, left: git2::Oid, right: git2::Oid) -> Result<Fork, git2::Error> {
+        if left == right {
+            return Ok(Fork::Parity);
+        }
+
+        let left_path = self.is_in_ancestry_path(left, right)?;
+        let right_path = self.is_in_ancestry_path(right, left)?;
+
+        if left_path && right_path {
```

Comment:
This is the same as `left == right`.

We can avoid grinding through the history by returning early if `left_path` is `true`, else if `right_path`. If both are `false`, we have a fork. 

There isn't really a way to tell which side was forked, because there is no main side here.